Automattic/wp-calypso
# Activity Log: Use Activity Log v2 in all versions of Calypso

-----

***NOTE:** Before this PR is merged, we'll need to add an illustration for the Activity Log's "empty" state. I've contacted our Design team for help coming up with one, but for now I've added a placeholder that just displays the string "Empty!" I'm happy to allow review, but I'll wait to deploy until the empty state is fully polished.*

-----

Since Jetpack Cloud went live, we've used two different versions of the Activity Log: the existing version for Calypso Blue, and a slight re-imagining of it in Calypso Green. This PR is a proposal for us to eliminate the older Calypso Blue version and replace it with the one currently used in Calypso Green. As a result, this PR would:

* reduce the surface area for bugs;
* make Calypso's Activity Log easier to maintain;
* unify the Activity Log experience between the two Calypsos; and
* significantly reduce the size of the bundle where the code for the Activity Log lives, improving performance on any page that references it.

#### Changes proposed in this Pull Request

##### The big stuff

* Modify `ActivityCardList` so that it always uses `ActivityCard` instead of `ActivityLogItem`, removing any related feature flag checks.
* Account for the "empty" state (i.e., when a site has no activities logged) by showing a relevant message on the Activity Log page.
* Move the "v1" Activity Log components that are still used in a specific scope, so that they're inside that scope in the file hierarchy: `ActivityLogItem`, `ActivityLogConfirmDialog`, and `ActivityLogRewindToggle`.

##### Janitorial stuff that follows as a result of the big stuff

* Remove the `activity-log/v2` feature flag from all config files.
* Remove all un-referenced "v1" Activity Log components: `ActivityLog`, `ActivityLogBanner`, `ActivityLogSwitch`, `ActivityLogTaskList`, and others.
* Remove the `ActivityLogExample` component, since `ActivityLogItem` is no longer meant to be widely used.

#### Testing instructions

1. Check out this PR (`git checkout remove/old-activity-log`) and start Calypso Blue (`yarn start`).
2. Select any Jetpack site. Verify the Activity Log works exactly as it does on Jetpack Cloud now, with no loss of functionality over the "original" Calypso Blue Activity Log.
3. Select any WordPress.com-hosted site (Atomic, Simple, etc.). Verify the Activity Log works for these sites as well, with no loss of functionality over the existing Calypso Blue Activity Log implementation.
4. Select a site with no logged activities (e.g., a site in "Coming soon" mode; a domain-only site; a site that's existed without updates for long enough that we would no longer show those activities on the page).

#### Reference screenshots

##### Before / After

<img width="350" src="https://user-images.githubusercontent.com/670067/166469749-cc1eedf4-b477-4f8e-91b0-99f004bf1e3e.png"> <img width="350" src="https://user-images.githubusercontent.com/670067/166469521-f6bff69c-76aa-4800-a65a-544158706d3f.png">

##### Empty state

<img width="350" src="https://user-images.githubusercontent.com/670067/166486040-fda2862d-ea3c-4e87-b851-a4454df5ab6b.png">
